src/inspector/inspector_socket_server.h

```diff
@@ -81,7 +81,9 @@ class InspectorSocketServer {
   const std::string host_;
   int port_;
 
+  std::mutex mutex_tcp_server_;
   std::shared_ptr<tcp_server> tcp_server_;
+  bool tcp_server_stopped_{false};

```

Comment:
>tcp_server_stopped_ [](http://example.com/codeflow?start=7&length=19)

Can we instead rely on the ServerState state_ field and protect it by the mutex too? Otherwise, we are effectively defining two different fields that keep the state of the server which are not synchronized with each other.

Reply:
Though, the keeping the separate bool tcp_server_stopped_ can be still simpler as long as we check the state_ variable in Start before we change it.

Reply:
I'll keep them separate. The object is already synchronously used and guaranteed to never Stop before Starting.

The only issue we need to address is with the tcp_server member which runs on a separate thread and in UTs we can actually call Stop before tcp_server's thread has a chance to get scheduled and start (would likely never happen in real use scenarios).